apps/labs/pages/api/annual-report.pdf.ts

```diff
@@ -0,0 +1,37 @@
+import type { NextRequest } from 'next/server';
+
+export const config = {
+  runtime: 'experimental-edge',
+};
+
+export default async function handler(req: NextRequest) {
+  // TODO - remove after checking the logs on Vercel
+  console.log(`
+  req.url ${req.url}
+  req.referer ${req.headers.get('referer')}
+  req.ip ${req.ip}
+  user-agent ${req.headers.get('user-agent')}
+  `);
+  // The shape of this fetch comes from the Plausible docs:
+  // https://plausible.io/docs/events-api
+  await fetch('https://plausible.io/api/event', {
+    method: 'POST',
+    headers: {
+      'User-Agent': req.headers.get('user-agent'),
+      'X-Forwarded-For': req.ip,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      domain: 'gabalafou.com',
+      name: 'pageview',
+      url: req.url,
+      referrer: req.headers.get('referer'),
+    }),
+  });
+  // Returning fetch comes from Next.js docs:
+  // https://nextjs.org/docs/api-routes/edge-api-routes#forwarding-headers
+  return fetch(
```

Comment:
Returning the PDF (rather than returning a redirect to the CDN URL to the PDF) in this Edge API Route might be the wrong way to do this. I just preferred the end user experience. You give the user a URL to the PDF, they go to it, and it loads under that URL, rather than redirecting and loading under the CDN URL. One effect of my decision is that if the user goes to copy the link, the link in their browser remains the same. (Which may or may not be desired from a marketing perspective.) 

For example, if I give an end user the following URL:

- labs.quansight.org/annual-reports/quansight-labs-annual-report-2022.pdf?utm_source=EthicalAds&utm_campain=AnnualReport

Then if the user copies the URL from their browser address bar and pastes it to a friend, it will be exactly the same URL.

My understanding is that doing it this way will increase our billable Vercel usage. That's because (if I understand correctly) Vercel's servers will be fetching and returning the PDF data through the URL route. But my understanding is that this route is temporary as we will eventually have an HTML page that contains the report, in which we can link directly to the CDN URL for the PDF and eventually delete this route altogether.

Reply:
Interesting 🤔 - @Katebrack is the `HTML` page referred to above a page where we will be linking to the various years' reports as in the NumFOCUS site?

Reply:
To be clear, we're not planning to create an HTML version and build out a page similar to NumFOCUS until ~next year. All we need for this year is the PDF link which, from what I read, we have

Reply:
> My understanding is that doing it this way will increase our billable Vercel usage.

How close are we coming to a breakpoint on our Vercel usage at present? If we're well within our current allocation (if we have one?), I don't think this is an issue.

On the other hand, if we *are* paying a pro rata on bandwidth usage, then we should consider resizing some images: https://vercel.com/dashboard/quansight/usage?topPathsFullscreen=true&pathType=source_path

Reply:
I prefer the UX this way, too.

Reply:
I changed it to use a redirect because that seems safer long-term.

Reply:
Causing the user's browser to change `.location` to the asset, instead of retrieving the asset server-side...which means that the users location bar will show the storyblok.com URL.

This clearly works, so I'm good with it functionally. What about the redirect makes it safer than the direct fetch?

Reply:
Safer in the face of ignorance. Vercel puts usage limits (timeouts, bandwidth) on these Edge API Routes functions. I assume that pumping the PDF through the handler would be fine, but this errs on the side of "I don't want to have to go fix it" if my assumption is somehow wrong.

Reply:
:+1:, the redirect makes it "more normal" traffic, and less likely to make Vercel cranky. Got it.

Reply:
Yeah, it's like I'm 99.9% sure the redirect is a safe and sound way to do this, versus I'm like 90% sure about the other way. And I've already put so much time into this, and there are still some other unverified assumptions that need to be checked, so this is me trying to not create anymore work for myself as well as giving myself some piece of mind.

Reply:
Yeah, I could see the fetch working great during our testing... and then once it goes live and a few hundred people (hopefully!) hit it over a few days Vercel then going "Nopers!" and kablooey.